lib/puppet/gettext_config.rb

```diff
@@ -0,0 +1,85 @@
+module Puppet::GettextConfig
+  LOCAL_PATH = File.absolute_path('../../locales', File.dirname(__FILE__))
+  POSIX_PATH = File.absolute_path('../../../../share/locale', File.dirname(__FILE__))
+  WINDOWS_PATH = File.absolute_path('../../../../../../puppet/share/locale', File.dirname(__FILE__))
+
+  # Search for puppet gettext config files
+  # @return path to the config, or nil if not found
+  def self.puppet_locale_path
+    if File.exist?(LOCAL_PATH)
+      return LOCAL_PATH
+    elsif Puppet::Util::Platform.windows? && File.exist?(WINDOWS_PATH)
+      return WINDOWS_PATH
+    elsif !Puppet::Util::Platform.windows? && File.exist?(POSIX_PATH)
+      return POSIX_PATH
+    else
+      nil
+    end
+  end
+
+  # Determine which translation file format to use
+  # @param conf_path the path to the gettext config file
+  # @return :mo if in a package structure, :po otherwise
```

Comment:
Not strictly necessary, but it can be helpful to include type info in these tags, ie
```
@return [Symbol] will be :mo if in a package structure, :po otherwise
```